sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AdaptiveSparkPlanExec.scala

```diff
@@ -297,17 +315,26 @@ case class AdaptiveSparkPlanExec(
         // plans are updated, we can clear the query stage list because at this point the two plans
         // are semantically and physically in sync again.
         val logicalPlan = replaceWithQueryStagesInLogicalPlan(currentLogicalPlan, stagesToReplace)
-        val (newPhysicalPlan, newLogicalPlan) = reOptimize(logicalPlan)
+        val (reOptimizationPhysicalPlan, newLogicalPlan) = reOptimize(logicalPlan)
```

Comment:
I think `reOptimize` should return `(Seq[SparkPlan], LogicalPlan)`, and here we pick the spark plan with the least cost between the current spark plan and newPhysicalPlans.

Reply:
If the conf is false, the returned `Seq[SparkPlan]` will be Nil, so that we don't introduce extra overhead if force skew join is disabled.